vtest/simple_test.go

```diff
@@ -0,0 +1,66 @@
+package vtest_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/suborbital/vektor/vk"
+	"github.com/suborbital/vektor/vlog"
+	"github.com/suborbital/vektor/vtest"
+)
+
+type simpleStruct struct {
+	Name string `json:"name"`
+	Age  int    `json:"age"`
+}
+
+func handleHello(r *http.Request, ctx *vk.Ctx) (interface{}, error) {
+	return vk.R(200, "hello"), nil
+}
+
+func handleSimpleStruct(r *http.Request, ctx *vk.Ctx) (interface{}, error) {
+	return vk.R(200, simpleStruct{"Bob", 30}), nil
+}
+
+func handleSetHeaders(r *http.Request, ctx *vk.Ctx) (interface{}, error) {
+	ctx.RespHeaders.Set("X-VK-TEST", "test")
+	ctx.RespHeaders.Set("X-SUBORBITAL", "rocket launch")
+
+	return vk.R(200, ""), nil
+}
+
+func TestVtest(t *testing.T) {
+	// suppress logging
+	logger := vlog.Default(vlog.Level(vlog.LogLevelError), vlog.ToFile("/dev/null"))
+
+	server := vk.New(
+		vk.UseLogger(logger),
+		vk.UseTestMode(true),
```

Comment:
I think we should consider switching back to having `server.TestStart()` instead of passing in `UseTestMode` as an option (or we could have both)... the reason is that I should be able to have a `createServer()` function in my codebase that returns the exact same server I use in production, and I should be able to do `vtest.New` from that without changing my production code to sometimes pass in `UseTestMode`.

Reply:
That makes sense to me. I don't see much use for a separate Option in that case though.